src/pages/Researchers/ProgramPostForm.js

```diff
@@ -0,0 +1,259 @@
+import React, { Fragment, useState } from "react";
+import { useNavigate } from "@reach/router";
+import { makeStyles } from "@material-ui/core/styles";
+
+import { Field, Form, FormSpy } from "react-final-form";
+import Grid from "@material-ui/core/Grid";
+import MenuItem from "@material-ui/core/MenuItem";
+import { notification } from "antd";
+import { required } from "../../modules/form/validation";
+import { categories, locations } from "../ProgramsList/ProgramItemsSider";
+import AppForm from "../../modules/views/AppForm";
+import Typography from "../../modules/components/Typography";
+import RFTextField from "../../modules/form/RFTextField";
+import FormFeedback from "../../modules/form/FormFeedback";
+import FormButton from "../../modules/form/FormButton";
+import AppFooter from "../../modules/views/AppFooter";
+
+const allTags = [...categories, ...locations];
+const programType = ["Survey", "Activity", "Others"];
+notification.config({ placement: "bottomRight" });
+const useStyles = makeStyles((theme) => ({
+  form: {
+    marginTop: theme.spacing(6),
+  },
+  button: {
+    marginTop: theme.spacing(3),
+    marginBottom: theme.spacing(2),
+  },
+  feedback: {
+    marginTop: theme.spacing(2),
+  },
+}));
+
+/**
+ * Creates a form for researchers to post a new program.
+ * @returns {*}
+ * @constructor
+ */
+const ProgramPostForm = () => {
+  const classes = useStyles();
+  const navigate = useNavigate();
+  const [sent, setSent] = useState(false);
+  const [selectedTags, setSelectedTags] = useState([]);
+
+  const validate = (values) => {
+    return required(
+      [
+        "title",
+        "date",
+        "venue",
+        "duration",
+        "compensation",
+        "type",
+        "number",
+        "description",
+      ],
+      values
+    );
+  };
+
+  const handleTags = (event) => {
+    event.persist();
+    setSelectedTags(
+      event.target.type === "checkbox"
+        ? event.target.checked
+        : event.target.value
+    );
+  };
+
+  const handleSubmit = (values) => {
+    const {
+      title,
+      date,
+      venue,
+      duration,
+      compensation,
+      type,
+      number,
+      description,
+    } = values;
+
+    try {
+      console.log(values);
+      console.log(selectedTags);
```

Comment:
let's clean this up afterwards if not needed. if needed for logging, log more description so log readers will have the context :)

Reply:
okay!